src/store/modules/app.ts

```diff
@@ -153,14 +162,31 @@ const actions = {
   getProposal: async ({ commit }, { space, id }) => {
     commit('GET_PROPOSAL_REQUEST');
     try {
-      const provider = getProvider(space.network);
       console.time('getProposal.data');
+      const response = await Promise.all([ipfsGet(gateway, id)]);
+      console.timeEnd('getProposal.data');
+      let [proposal]: any = response;
+      proposal = formatProposal(proposal);
+      proposal.ipfsHash = id;
+
+      commit('GET_PROPOSAL_SUCCESS');
+      return { proposal };
+    } catch (e) {
+      commit('GET_PROPOSAL_FAILURE', e);
+    }
+  },
+
+  getResults: async ({ commit }, { space, id }) => {
+    commit('GET_RESULTS_REQUEST');
+    try {
+      const provider = getProvider(space.network);
+      console.time('getProposal.results');
       const response = await Promise.all([
         ipfsGet(gateway, id),
```

Comment:
We shouldn't load again the proposal data, we got it already previously we should reuse it.